src/integrationtests/java/com/aws/greengrass/integrationtests/ShadowManagerTest.java

```diff
@@ -247,4 +264,63 @@ void GIVEN_shadow_manager_WHEN_strategy_config_resets_THEN_respond_to_config_upd
     }
+
+    @Test
+    void GIVEN_shadow_manager_running_AND_crypto_service_loading_WHEN_sync_requests_with_dataplane_client_THEN_retry_requests_with_good_client_after_crypto_service_loads(ExtensionContext context)
+            throws InterruptedException, TLSAuthException, IoTDataPlaneClientCreationException {
+        ignoreExceptionOfType(context, TLSAuthException.class);
+        ignoreExceptionOfType(context, RetryableException.class);
+
+        SecurityService securityService = mock(SecurityService.class);
+        when(securityService.getDeviceIdentityKeyManagers()).thenThrow(TLSAuthException.class);
+        kernel.getContext().put(SecurityService.class, securityService);
+
+        IotDataPlaneClientFactory factory = kernel.getContext().get(IotDataPlaneClientFactory.class);
+        IotDataPlaneClientWrapper wrapper = spy(new FakeIotDataPlaneClientWrapper(factory));
+        kernel.getContext().put(IotDataPlaneClientWrapper.class, wrapper);
+
+        // Shadow manager starts syncing 1 shadow in the realtime as per the recipe and getThingShadow
+        // cloud api is called as part of syncing.
+        startNucleusWithConfig(NucleusLaunchUtilsConfig.builder()
+                .configFile("sync.yaml")
+                .mqttConnected(true)
+                .mockCloud(false)
+                .build());
+        BaseSyncStrategy syncStrategy = kernel.getContext().get(RealTimeSyncStrategy.class);
+        assertThat("syncing has started", syncStrategy::isSyncing, eventuallyEval(is(true)));
+        verify(wrapper, timeout(5000).atLeast(1)).getThingShadow("Thing1", "");
+        verify(securityService, timeout(5000).atLeast(3)).getDeviceIdentityKeyManagers();
+
+        // This test ensures that the request is retried again as the we're not mocking cloud and data plane client
+        // creation fails. If the data plane client creation succeeds, the request is processed once again.
+        CountDownLatch cdl = new CountDownLatch(1);
+        reset(securityService);
+        when(securityService.getDeviceIdentityKeyManagers()).thenAnswer((invocation) -> {
+            cdl.countDown();
+            return new KeyManager[0];
+        });
+        assertThat("request is retried with a new client", cdl.await(10, TimeUnit.SECONDS), is(true));
+        verify(securityService, atMostOnce()).getDeviceIdentityKeyManagers();
+        verify(wrapper, timeout(5000).atLeast(2)).getThingShadow("Thing1", "");
+        // Once the request is processed, the sync queue should be empty.
+        assertEmptySyncQueue(RealTimeSyncStrategy.class);
+    }
+
+    class FakeIotDataPlaneClientWrapper extends IotDataPlaneClientWrapper {
+        IotDataPlaneClientFactory factory;
+
+        public FakeIotDataPlaneClientWrapper(IotDataPlaneClientFactory iotDataPlaneClientFactory) {
+            super(iotDataPlaneClientFactory);
+            factory = iotDataPlaneClientFactory;
```

Comment:
```suggestion
```

Reply:
The parent is already storing the factory on a variable

Reply:
`iotDataPlaneClientFactory` is a private field in the wrapper. So, I added this to use the injected factory.